types/block_test.go

```diff
@@ -10,7 +10,25 @@ import (
 	cmn "github.com/tendermint/tendermint/libs/common"
 )
 
-func TestValidateBlock(t *testing.T) {
+func TestBlockAddEvidence(t *testing.T) {
+	txs := []Tx{Tx("foo"), Tx("bar")}
+	lastID := makeBlockIDRandom()
+	h := int64(3)
+
+	voteSet, valSet, vals := randVoteSet(h-1, 1, VoteTypePrecommit, 10, 1)
+	commit, err := MakeCommit(lastID, h-1, 1, voteSet, vals)
+	require.NoError(t, err)
+
+	block := MakeBlock(h, txs, commit)
+	require.NotNil(t, block)
+
+	ev := NewMockGoodEvidence(h, 0, valSet.Validators[0].Address)
+	block.AddEvidence([]Evidence{ev})
```

Comment:
shouldn't we finish the test by checking that the evidence got added?

Reply:
yes